src/Service/CreatePluginClass.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+declare(strict_types=1);
+
+namespace PharIo\Mediator\Service;
+
+use SplFileInfo;
+use function file_exists;
+use function file_put_contents;
+
+final class CreatePluginClass
+{
+	private $targetDirectory;
+
+	public function __construct(SplFileInfo $targetDirectory)
+	{
+		$this->targetDirectory = $targetDirectory;
+	}
+
+	public function __invoke(string $namespace): void
+	{
+		$class = <<<EOF
+<?php
+
+declare(strict_types=1);
+
+namespace $namespace;
+
+use PharIo\ComposerDistributor\ConfiguredMediator;
+
+final class Plugin extends ConfiguredMediator
+{
+    public function getDistributorConfig(): string
+    {
+        return __DIR__ . '/../mediator.xml';
+    }
+}
+
+EOF;
```

Comment:
Do we need to have this inline or can we put that into a "template" type of file?

Reply:
The new change uses the already available `Plugin.php` file as template. The code is read on instantiating `CreatePluginClass`  so that the template class can be removed without affecting the class.